src/network.py

```diff
@@ -33,7 +33,8 @@ def build_msdnet(
             convolution=convolution
             )
     else:
-        dilation_array = np.array(msdnet_parameters.dilation_array)
+        dilation_array = [int(x) for x in msdnet_parameters.dilation_array.strip('[]').split(', ')]
```

Comment:
I think splitting on `','`, and then stripping each item would be a bit more lenient and more in line with Python allowing spaces between list items as well.

Reply:
I am still not quite sure why the proposed order will be more pythonic/general, but I have made changes accordingly.

How does this look like?
`dilation_array = [int(x.strip('[]')) for x in msdnet_parameters.dilation_array.split(', ')]`

For the sake of time, I will go ahead and make changes to other places in the code where we have similar parameters, and happy to discuss more.

Reply:
Not what I meant :)

`dilation_array = [int(x.strip()) for x in msdnet_parameters.dilation_array.strip('[]').split(',')]` allows for both `[1, 2, 3]` and `[1,2,3]`

Reply:
I think the `x.strip()` for each element is not necessary, the conversion automatically handles the white space.

I have tried `dilation_array = [int(x) for x in msdnet_parameters.dilation_array.strip('[]').split(',')]` with:
`'[1, 2, 3]'`, `'[1,2,3]'` and `'[1,            2,         3]'`, all works.